Add openstack/esi integration roles

larsks opened the pull request:
Add roles to set up floating ip addresses, port forwarding, and dns records for cluster api and ingress endpoints.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Expanded cloud automation support with new integration collections and dependencies: `openstack.cloud`, `amazon.aws`, `community.general`, and `ansible.utils`.
  - Introduced dynamic role-based workflows to automate the creation and deletion of hosted clusters.
  - Enhanced configuration options for environment variables, network settings, and DNS record management.
  - Added support for managing ingress infrastructure within Kubernetes environments.

- **Chores**
  - Streamlined processes for floating IP allocation and removal.
  - Upgraded common infrastructure settings to improve resource provisioning and overall operational reliability.
  - Customized linting behavior by adding specific entries to ignore certain rules.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

adriengentil commented on playbook_cloudkit_create_hosted_cluster.yml:
```diff
@@ -14,11 +16,25 @@
         hosted_cluster_settings: "{{ cluster_settings_result }}"
         hosted_cluster_state: present
+
+    - role: infrastructure/api
```
generally, `ansible-lint` doesn't really like nested roles. Maybe the solution would be to split the roles in collections, e.g.: 
- `collections/cloudkit/hosted_cluster/roles/...`
- `collection/cloudlkit/infrastructure/roles/...`

Last time I tried it failed and I did not look for a solution, but I could be a better way to organize our stuff

larsks replied:
I have configured ansible-lint to disable the `role-name[path]` rule for now. I've updated everything to either fix the linting errors or to add `noqa:...` comments where I disagree with `ansible-lint`.

larsks replied:
Sure, let's merge this and then we can work on restructuring it.

adriengentil replied:
Created https://github.com/innabox/cloudkit-aap/issues/18 to track it

adriengentil commented:
@larsks ansible-lint complains about:
```
couldn't resolve module/action 'kubernetes.core.k8s'. This often indicates a misspelling, missing collection, or incorrect module path.
````

because we removed `collections/requirements.yml`. ansible-lint uses this file to resolve the dependencies. Maybe it's better to restore it?

larsks commented:
> because we removed collections/requirements.yml

Not intentionally. I changed it into a symlink, but apparently forgot to include it in a commit. Pushing an update with that fix.
